### PR TITLE
Delete source link category

### PR DESCRIPTION
**Pull request by MishaHMK**
dev

## Summary of issue

[Issue](https://github.com/orgs/project-studying-dotnet/projects/15/views/1?pane=issue&itemId=79455193)

## Summary of work

**CREATED:**
Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Delete/ - DeleteCategoryQuery (query)
Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Delete/ - DeleteCategoryHandler (handler)
Streetcode/Streetcode.XUnitTest/MediatRTests/Sources/SourceLinkCategory/ - DeleteCategotyHandlerTests (unit-tests)

**MODIFIED:**
Streetcode/Streetcode.WebApi/Controllers/Source/ - SourcesController (delete action)

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=50%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Sources/SourceLinkCategory/DeleteCategotyHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+            // Act
+            var result = await _handler.Handle(query, CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsSuccess);  
+        }
+
+        [Fact]
+        public async Task Handle_ReturnsSuccess_WhenCategoryNotFound()
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Sources/SourceLinkCategory/DeleteCategotyHandlerTests.cs:57]: Test for failure result () named `..._ReturnsSuccess_...`

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Sources/SourceLinkCategory/DeleteCategotyHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+using Streetcode.BLL.MediatR.Streetcode.RelatedTerm.Delete;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+
+using SourceLinkCategoryEntity = Streetcode.DAL.Entities.Sources.SourceLinkCategory;
+
+namespace Streetcode.XUnitTest.MediatRTests.Sources.SourceLinkCategory
+{
+    public class DeleteCategotyHandlerTests
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Sources/SourceLinkCategory/DeleteCategotyHandlerTests.cs:17]: Typo 'CategoTy'.

File: Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Delete/DeleteCategoryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                return Result.Fail(errorMsg);
+            }
+
+            _repository.SourceCategoryRepository.Delete(srcCategory);
+            var resultIsSuccess = await _repository.SaveChangesAsync() > 0;
+            var srcCategoryDto = _mapper.Map<SourceLinkCategoryDto>(srcCategory);
+
+            if (resultIsSuccess && srcCategoryDto != null)
+            {
+                return Result.Ok(srcCategoryDto);
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Delete/DeleteCategoryHandler.cs:41]: Why do you need to return Dto deleted item?

**Review by jimmy2jk — CHANGES_REQUESTED** [review]

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Sources/SourceLinkCategory/DeleteCategoryContentByStreetcodeIdHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+             .ReturnsAsync(sourceLinkCategoryContent);
+
+            _repositoryMock.Setup(r => r.SaveChangesAsync()).ReturnsAsync(1);
+
+            // Act
+            var result = await _handler.Handle(query, CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsSuccess);
+            _repositoryMock.Verify(r => r.SaveChangesAsync(), Times.Once);
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Sources/SourceLinkCategory/DeleteCategoryContentByStreetcodeIdHandlerTests.cs:49]: Verify call of Delete method. Not SaveChangesAsync

**Review by jimmy2jk — CHANGES_REQUESTED** [review]

**Review by jimmy2jk — APPROVED** [review]
Great Job👍👍👍

**Review by refokcer — APPROVED** [review]

**Pull request merged**